### DISCUSSION TITLE
Jupyter process still running after close the notebook file

### DISCUSSION THREAD

**Discussion by PolarisRisingWar** [microsoft/vscode-jupyter#11074] [Questions and Answers]
I can directly use ipynb file to run jupyter code. But after I closed the file, the processes about jupyter are still running in the background (by using command line `ps -aux | grep jupyter` I can find a lot of them.  
Is that a normal situation? Or I missed some bugs?

**Comment by DonJayamanne** [comment]
> Is that a normal situation? Or I missed some bugs?

unfortunately this is a bug.
please could you 
* Send a list of the process that are left running.
* shut down so the processes, opens s notebook and run a cell
* close the notebook and check if any processes are still running and provide that list again

**Comment by PolarisRisingWar** [comment]
1. I closed all the notebook files in VSCode editor.
2. By running `ps -aux | grep jupyter`, I listed all my processes (coz this remote server has more than one user) about notebook, and killed one of them, and it showed this message:

![image](https://user-images.githubusercontent.com/48322321/183568202-928a21c3-e422-4050-b6b9-34dbe1d150d0.png)
3. I reopened a notebook file, and I can normally run codes.
4. I killed all of them, and at once it showed similar messages again.
5. I closed all the notebook files, and reopened a notebook file, and I can't directly run codes. And I restarted the kernel, and I can run the code. Then I saved and closed the file. And again, I found the process is still here, and then I manully kill this process, again, it showed me this message:
![image](https://user-images.githubusercontent.com/48322321/183572140-e01dbbab-c725-47f3-b643-a0a23349201a.png)

**Reply by DonJayamanne** [reply]
thanks, im after a screenshot of the running processes.
please let `ps -aux | grep jupyter`

please note, Jupyter will be shut down only when you close vscode.

**Reply by PolarisRisingWar** [reply]
How about remote server? I'm using remote server and I could see those jupyter processes with the time before I've closed remote connections in the past. Even after I closed remote connections, the jupyter server seems still running in the background.

**Reply by DonJayamanne** [reply]
we never shut down remote kernels nor the server. 
just like jupter itself, if you open a notebook and start a kernel you'll need to shut it down manually. we try to preserve that behavior with remote kernels

**Comment by PolarisRisingWar** [comment]
It seems like just closing the file, the server is still running?
